File: joiner-spring/src/test/java/cz/encircled/joiner/test/repository/UserRepositoryImpl.java
package cz.encircled.joiner.test.repository;

import cz.encircled.joiner.spring.SpringJoinerRepositoryImpl;
import cz.encircled.joiner.test.model.User;
import org.springframework.stereotype.Repository;

/**
 * @author Kisel on 21.01.2016.
 */
@Repository
public class UserRepositoryImpl extends SpringJoinerRepositoryImpl<User> implements UserRepository {

}

File: joiner-spring/src/test/java/cz/encircled/joiner/test/repository/UserRepositoryImpl.kt
package cz.encircled.joiner.test.repository

import com.mysema.query.types.EntityPath
import cz.encircled.joiner.spring.SpringJoinerRepository
import cz.encircled.joiner.test.model.User
import org.springframework.stereotype.Repository

/**
 * @author Kisel on 21.01.2016.
 */
@Repository
class UserRepositoryImpl : SpringJoinerRepository<User>(), UserRepository {

    protected override val rootEntityPath: EntityPath<User>?
        get() = QUser.user1

}
